### DISCUSSION TITLE
Love it!!

### DISCUSSION THREAD

**Discussion by erohtar** [adirh3/Fluent-Search#401] [General]
Been a happy Listary user for several years. Just found out about Fluent Search today, tested it out and love it so far! Just wanted to show some appreciation and let the developer know that it's amazing to see the rapid pace of development - shows how passionate you are about the project.

Though I do miss some Listary features like list of favorite folders available via hotkey anywhere and in a toolbar next to file selection dialogs, I'm hopeful those and other useful features will get added to FS in future.

**Comment by adirh3** [comment]
Hey @rajat-r, I am glad you liked Fluent Search!
Thanks for the feedback, other users have been asking for these features as well and I have been thinking a lot about how to properly implement them.

Some of these you can already achieve using a combination of existing features.
For example, you can set Hotkey to open FS with specific search tag (in Settings -> Power user) and make FS open with the `folder` search tag to show commonly used folders.
You can also add any results to a custom search tag using the Plus icon next to its tags, that way you can quickly access a list of configured search results using a hotkey.
![image](https://user-images.githubusercontent.com/27368554/148427432-d10cd329-e986-4a3a-baf8-996baef29798.png)
Hopefully these can help for the time being.

Feel free to join the Discord server - https://discord.gg/W2EuWvD6GD and share any feedback/ask any question there.

**Reply by erohtar** [reply]
Thanks for the reply @adirh3 ! 
Great to know there are options to implement those features, though adding more global hotkeys to my current windows setup would likely be counter-productive. Instead how about letting users use a tiny text as the first search term? eg a simple 'g' (followed by space/tab) to search google etc.

The above was also mentioned by another Listary user on their forum (that's where I found about FS). I know FS is not a Listary-clone, but some features just seem intuitive enough that adapting them as such might just be the best idea.

As an aside, I see you're active on issues posted here on Git, and you just suggested Discord - which one would you say you prefer and users are more likely to get heard on? Keeping it limited to a single platform may be a good idea for you too, so you don't spread yourself too thin.

Cheers!

**Reply by adirh3** [reply]
@rajat-r I might've confused you with the hotkey, I meant that it can be used to bring FS as a global "List" of configured results as you mentioned you wanted to show list of favorite folders via hotkey.

As for `Search tags`, let me explain -
In Fluent Search, `Search tags` are words assigned to results, allowing you to filter results. For example, for `Google` web results, the tags are `Google` and `Web` (used to search in all engines).
![image](https://user-images.githubusercontent.com/27368554/148434068-476f7c66-e582-4acc-9195-fc5fa1112ccc.png)

To use `Search tags` to filter results, you enter the tag name, then press `TAB` to insert it. FS will auto complete tags as you type, so you will only need to type `g` and then press `TAB` to insert the `Google` tab.
![image](https://user-images.githubusercontent.com/27368554/148432214-2e8af7ca-448f-43de-bd25-ed125716788c.png)

In Listary/Wox, as you know you use keywords, and to invoke them by typing them followed by space and then the search term. 
I decided to not go with this approach, here are the reasons - 
- It does not interfere with normal search, which can confuse users who are not familiar with Listary/Wox
- It is the same keys needed to be pressed as Listary/Wox - `g`, `Tab` and `g`, `Space`
- Its same mechanism used in Chrome/Edge, which many users are familiar with
- It allows inserting the tag **after** the search term, for example you search "myfolder" and then type "folder" and press TAB
- It allows `search tags` that contains space, such as folder paths, you can type full path of folder, then press `TAB` to search in that folder
- It allows you to use the full name instead of synonyms, for example `Google` instead of `g` because of auto completion, which is less confusing to users who are not familiar with Listary/Wox
- It allows inserting tags from the focused result using `Tab`, for example here pressing TAB will search inside this folder - 
![image](https://user-images.githubusercontent.com/27368554/148435032-4a9b6c18-e654-4fc2-9604-296c4c5e728a.png)

In addition, you can always change these search tags in Settings if you'd like (for example set `b` for `Google`).

As for Discord/GitHub, you're completely right about it, what I am trying to achieve is to be able to get as much feedback as possible but not necessarily as efficient as possible.
Users prefer different platforms, and when I notice an enthusiastic user like you, I let them know about all the platforms available to provide feedback.
In general, for bug reports I prefer to use Github, but for normal feedback the Discord is usually better as there are many features users might miss and asking on Discord will get you a faster response and not necessarily from me. 

Hopefully this clears up things, thanks again for your feedback!

**Comment by erohtar** [comment]
I see what you mean with that Google example, and while it does work great
in a lot of ways, it does slow one down - pls let me explain.
So as you know Listary/Wox's google search is 'g <searchterm>' but with FS,
when I press 'g{tab}' it's searching in game controllers:
![image](https://user-images.githubusercontent.com/5120628/148443752-9b1fea4c-7626-4f24-b214-6a35f7d8dc5c.png)


If I'm trying to look into processes, it goes through:
![image](https://user-images.githubusercontent.com/5120628/148443784-4f7acfca-fb82-45d0-8b6a-3d3a27cc41b5.png)

![image](https://user-images.githubusercontent.com/5120628/148443801-e4fdcba6-c4fc-4c9c-a782-077e0595970c.png)

![image](https://user-images.githubusercontent.com/5120628/148443809-0c57720e-05d8-4c86-8109-c8683c642def.png)


So, here I have two problems that slow me down:
1) Having to type 3 keys (could be more in another example) instead of just
1, and that's not efficient
2) And (this is even bigger than the one above) it slows me down by me
having to look at the results after each typed letter, just to make sure my
Tab will take me to the right search! That's a huge downside to this
approach IMO. And to make it worse, as more files/applications get added to
my computer, how many letters typed gets me to the right search keeps
changing - so no future muscle memory benefits either!

Someone pointed this out (agreed, not very nicely) in the Listary forum too:
![image](https://user-images.githubusercontent.com/5120628/148443831-3657b798-ce4f-4151-bf3c-638da7a8a3c5.png)

Edit: fixed images not working because I replied via email

**Reply by adirh3** [reply]
Thanks for the explanation @rajat-r.
Let me see if I understand - so the issue here is not about "TAB vs Space", but that FS **by default** shows some junk that interfere with the search tags, right?
Did you enable the `Suggest tags from results` setting?  Otherwise, you shouldn't see `Game controllers` suggested as in your screenshot.
By default (without touching settings) it should look like this - 
![image](https://user-images.githubusercontent.com/27368554/148445411-798e32e9-7ad6-464b-9f08-9b2761547365.png)

What I am aiming for is that by default FS tag system will offer only things necessary and with minimal key pressed, for example, web engines, the "apps", "processes", and "bookmarks" tags.
And if you'd like to further customize it, you can do it very easily in settings by setting `g` as search tag for `Google` web results.
![image](https://user-images.githubusercontent.com/27368554/148445800-4a12068a-17a5-4c81-aa1c-4d86f9415ae0.png)
Or even map **any** text to search tags in Settings -> Search tags -
![image](https://user-images.githubusercontent.com/27368554/148445511-d1a0994a-e93d-48ea-9e95-4db71f53c302.png)
I might as well set `g` as default tag for `Google` as other users might enable `Suggest tags from results`.

I'd like Fluent Search to have the most intuitive UX by default, as you initially used Fluent Search were the search tags confusing? If so, what would you change?
If we're setting aside the "TAB vs Space", by default FS should have the same experience as Listary, also as it allow users to customize the "search tags" as Listary let users sets keywords.
Thanks!

**Reply by erohtar** [reply]
Yes @adirh3, I had that enabled! I played a lot with the settings, so maybe I changed the default there.
Btw is there a central location in settings to see ALL the tags (and edit them) in one place?

Speaking of settings, I did try more than once to open FS Settings from within FS itself, but couldn't get it to work - is it not supported? A tag to directly go to a certain setting would be a great idea IMO.

And sorry about throwing a lot of things into this thread, but I found an issue with the "Instantly Search Processes". I enabled it with a couple apps, but it kept getting triggered incorrectly so I had to disable. An example is KeePass - when in the ListView it works, but when trying to open any of the menus via Alt+key combinations, FS pops up and grabs that key. 
Listary here shines bright - it just works in pretty much ALL ListViews in most apps and I've never come across such issues with it.

> I'd like Fluent Search to have the most intuitive UX by default, as you initially used Fluent Search were the search tags confusing? If so, what would you change?
If we're setting aside the "TAB vs Space", by default FS should have the same experience as Listary, also as it allow users to customize the "search tags" as Listary let users sets keywords.

Yes, it was confusing to me .. but how would I change it - let me think on that some more before I answer.

**Reply by adirh3** [reply]
@rajat-r currently there is no one place to see all tags, they are used in many `search apps` across Fluent Search.
It is set that each `search app` for example `Apps` has it's own tag, which can be disabled by disabling the app or you can search in it only using the tag -
![image](https://user-images.githubusercontent.com/27368554/148448787-421c36bd-a860-47e7-af41-0eb558b857ea.png)

I am thinking about having a convenient place for managing all tags, but still thinking about its implementation.

As for Instantly search processes, I will be honest this is a place where FS is still behind Listary.
This feature was 100% inspired by Listary users and I feel like I haven't fully managed all scenarios well as Listary does (For example Alt+key as you said).
One thing I am not sure I understand, you mention `KeePass` which is not a file manager from my understanding, how does Listary interact with it? I thought Listary "find as you type" feature works only with file managers, am I wrong?

**Reply by erohtar** [reply]
> I am thinking about having a convenient place for managing all tags, but still thinking about its implementation.

Okay great, as long as you have it on your radar.

> One thing I am not sure I understand, you mention KeePass which is not a file manager from my understanding, how does Listary interact with it? I thought Listary "find as you type" feature works only with file managers, am I wrong?

Listary also lets you search for text in most ListView controls. FS does it too, right? (or is it a side-effect of adding the feature for file explorers?)

But here's a simple example - this is a window of an application that I made in AutoHotkey, with a standard ListView control. I didn't have to do anything to Listary, but it just lets me search for values in here. 
![image](https://user-images.githubusercontent.com/5120628/148450732-d49636b2-54ed-46d8-88a9-bcfc7810f79d.png)

FS lets me do that too, but it does have that Alt+key issue. Also, (minor feature) when traveling through the list in Listary results, the actual entry in the ListView gets selected automatically, but not so in FS. Though I thought the "Interact with result on focus" setting did that, but it doesn't.

**Reply by adirh3** [reply]
Very cool I did not know Listary can do that, thought it was exclusive to file managers.
Yes, FS should be doing that as well and "Interact with result on focus" should be doing exactly that so it's a bug.
If you can please open a GitHub issue for these two bugs, once they're fixed in the nightly feed it will automatically notify you.

Thanks again!

**Reply by erohtar** [reply]
Sure thing, done!
